### DISCUSSION TITLE
How to debug problem reading geojson file

### DISCUSSION THREAD

**Discussion by claremacrae** [javalent/obsidian-leaflet#267] [Q&A] [answered]
I am trying to work out how to get this plugin to draw the contents of a GeoJson file.

Whatever I try, my map only ever looks like this:

![image](https://user-images.githubusercontent.com/4840096/147775595-39b0db2b-2f93-4888-9492-2baf8d92ef7f.png)

My Markdown file `leaflet_geojson_sample.md` contains this:

```
\```leaflet
id: leaflet_geojson_sample
geojson: [[leaflet_geojson_sample.json]]
height: 502px
lat: -87.617232
long: 41.8784146
minZoom: 5
maxZoom: 10
defaultZoom: 6
unit: meters
scale: 0.1
darkMode: false
marker: default, 87.617232, 41.8784146, [[leaflet_geojson_sample]]
```

My `leaflet_geojson_sample.json` file:

- was cribbed from https://github.com/valentine195/obsidian-leaflet-plugin/discussions/124
- works fine on [geojson.io](https://geojson.io/#map=8/0.500/102.502)
- contains this:

```json
    {
      "type": "Feature",
      "properties": {
        "title": "Jackson Road Extension",
        "description": "Extension of US 66 in 1937, upon completion of land reclamation for 1937 Chicago World's Fair",
        "stroke": "#ff0000",
        "stroke-width": 3
      },
      "geometry": {
        "type": "LineString",
        "coordinates": [
          [
            -87.617232,
            41.8784146,
            0
          ],
          [
            -87.6243238,
            41.8783107,
            0
          ]
        ]
      }
    }
```

**Comment by valentine195** [comment]
Hmm, a GeoJSON file shouldnt prevent map tiles from loading. What version of the plugin are you on and are there any console errors?

**Reply by claremacrae** [reply]
Thank you - I updated this afternoon to 4.8.0

Several other maps work fine.

**Reply by claremacrae** [reply]
Console shown below - there are no messages from this plugin

**Reply by valentine195** [reply]
Okay, I’ll have to take a look at this one specifically later tonight and see what is going on. 

You could add `verbose: true` to the map parameters to get more logging info but I doubt it’ll help with this.

**Reply by claremacrae** [reply]
Oooh - I hadn't noticed that option. It adds this output:

```
Obsidian Leaflet Map leaflet_geojson_sample: Beginning Markdown Postprocessor.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Loading immutable items.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: MarkdownRenderChild loaded. Appending map.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Found 1 markers and 0 overlays.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Loading GeoJSON files.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Getting initiatial coordinates.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Beginning render process.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Building initial map layer.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Adding 1 GeoJSON features to map.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: 1 GeoJSON feature added to map.
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Moving to supplied coordinates: -87.617232,41.8784146
VM133:2 Obsidian Leaflet Map leaflet_geojson_sample: Initial map layer rendered in 0.024 seconds.
﻿```

**Reply by claremacrae** [reply]
Does the plugin cache any data anywhere outside of the vault directory?

**Reply by claremacrae** [reply]
With Obsidian closed, I duplicated the pair of files to `leaflet_geojson_sample_two` - with a global replace inside the mark-down file...

And then restarted Obsidian. This was to see if there were any problems due to remnants from the earlier mistake.

I got the same blank map.

**Reply by valentine195** [reply]
It does not, no. All data is reloaded when the map is rendered.

**Reply by claremacrae** [reply]
Thanks - I asked because, with a map that contains `markerFolder: Clare/Volunteering/CSF Locations`,  I often find I have to edit some map setting in the markdown file containing the leaflet block, to get it to include updated data, having edited some locations in files in that folder... So it wasn't automatically reloading data from elsewhere in Obsidian...

**Reply by claremacrae** [reply]
But with the absence of a cache, that's a different topic to this one...

**Reply by valentine195** [reply]
Hmm, the map should detect updates to files and update itself. If it’s not doing that, it’s a bug.

**Reply by claremacrae** [reply]
I wasn't going to mention this, but by adding the verbose flag you mentioned, I can definitely show that the map does not update when underlying data in other files is updated.

But there is a workaround to that, which is to edit a parameter in the leaflet block, whereas I've been unable to find a workaround to the one in this ticket, so am focussing on that!

**Comment by claremacrae** [comment]
In an earlier version of the markdown file, there was this line

```
gpx: [[leaflet_geojson_sample.json]]
```

instead of 

```
geojson: [[leaflet_geojson_sample.json]]
```

So I can definitely understand why that didn't work....

What I cannot figure out is how to see any error messages associated with the plugin's parsing of data.

The Obsidian 0.13.15 console has this:

![image](https://user-images.githubusercontent.com/4840096/147776097-c98a9f57-fd30-418d-94b0-182aed30320d.png)

**Comment by claremacrae** [comment]
I turned my computer off now, but playing with this file on my iPad, I noticed that I had a marker at latitude +87, but the map was entered on latitude -87 - either way, there's lots of white background, which may have made it look like the map didn't load!

I think it's likely that when I copied the coordinates form the geojson file, I didn't notice that the list and long may have been the other way round...

So there's a reasonable chance that this is user error, or user not yet knowing...

I'll test it out tomorrow and let you know...

The bit in my question about "how could I debug this?" is still valid, though, I think...

**Comment by claremacrae** [comment, marked as answer]
Yeah, tested on iPad and got it working... So the error was in my coordinates in the markdown leaflet block, as I mentioned in the message above.

With this Markdown, it works fine and loads the line from the geojson posted above...

```
\```leaflet
id: leaflet_geojson_sample
geojson: [[leaflet_geojson_sample.json]]
height: 502px
lat: 41.8784146
long: -87.62432
minZoom: 5
maxZoom: 20
defaultZoom: 15
unit: meters
scale: 0.1
darkMode: false
verbose: true
marker: default, 41.8785146, -87.62432, [[leaflet_geojson_sample]]
```

So sorry for the red herring, but I spent hours on this before asking the question here, so having a way to track down errors like this would be really, really helpful. even if just more info in the verbose output, like the lat+long of the bounding box displayed, and any similar information... If I'd seen that my coords were different ways round, I'd have spotted the problem sooner, I suspect.

Anyways, thanks again for this awesome plugin!

**Reply by valentine195** [reply]
If you can think of some logging I can add to the verbose mode that would help with this situation, I can definitely add it no problem.

**Reply by claremacrae** [reply]
Without knowing about the implementation, it's hard to say... What types of values does the code calculate, to decide what to display?
Does it calculate any bounding boxes, like what are the limits of the visible area?
Perhaps:
Displaying from(lat+long at bottom-left to lat+long at top-right)
Any display would ideally be in the units specified by the user. So that the values printed can be compared with values in the input files.
It's really hard for me to say without spending ages reading the code.
But I'm hope you will have an idea of the kinds of things that users will have go stuck on, from the support questions you get asked.

example: If my marker is missing, is it because it is out of the displayed area, or did it get lost for some reason, like my settings being lost?
A log message like this would help - :
- 0 markers read from settings for the map with id 'blah'
- 37 things (I don't know the vocabulary) loaded from geojson file
- 25 markers read from files in folder 'a/b/c'
